modules/thermo-cycler/thermo-cycler-arduino/lid.cpp

```diff
@@ -139,9 +186,8 @@ void Lid::_update_acceleration()
   _calculate_step_delay();
 }
 
-void Lid::move_millimeters(float mm, bool top_switch, bool bottom_switch)
+bool Lid::move_millimeters(float mm)
```

Comment:
I think there's an issue here. This function is intended to loop until either
- it hits the specified distance
- the appropriate switch is triggered

But, I don't think it can sense the switch. The switch state is updated in `Lid::switch_poller`, which is called synchronously from `loop()` when the timer interrupt sets a flag. However, `move_millimeters` is _also_ called synchronously from loop. So `_is_cover_switch_pressed` and `_is_bottom_switch_pressed` can't actually be updated while this function is happening. This function looks like it will move the full distance every time and ignore the switches.